src/widget/form/profileform.cpp

```diff
@@ -416,3 +421,31 @@ void ProfileForm::retranslateUi()
     // We have to add the toxId tooltip here and not in the .ui or Qt won't know how to translate it dynamically
     toxId->setToolTip(tr("This bunch of characters tells other Tox clients how to contact you.\nShare it with your friends to communicate."));
 }
+
+void ProfileForm::onRegisterButtonClicked()
+{
+    QString name =  bodyUI->dnsUsername->text();
+    QString id = toxId->text();
+    QString bio = bodyUI->bio->text();
+    QString server = bodyUI->serversList->currentText();
+    bool privacy = bodyUI->privacy->isChecked();
+    if (name.isEmpty())
+        return;
+
+    QString response = Toxme::createAddress(server, id, name, privacy, bio);
+
+    bool ok;
+    int errorCode = response.toInt(&ok);
+    if (ok)
+    {
+        QString errorMessage = Toxme::getErrorMessage(errorCode);
+        QMessageBox::warning(this, tr("Toxme error"),  errorMessage, "Ok");
+    }
+    else
+    {
+        QMessageBox::information(this, tr("Done!"), tr("Successfully added %1@%2 to the database. Save your password").arg(name, server), "Ok");
+        bodyUI->password->show();
+        bodyUI->passwordLabel->show();
+        bodyUI->password->setText(response);
```

Comment:
Sorry. I don't understand. What i must do here?
// did we time out or QIODevice got closed?